make/test-math-dependencies

```diff
@@ -24,7 +24,10 @@
 ##
 define math_warnings
 $(eval type:= $1)
-$(eval all_headers := $(shell find $2 -name '*.hpp' -o -name '*.cpp'))
+$(eval excl:=)
+$(eval excl_ids:= 1 2 3 4 5)
+$(foreach a,$(excl_ids),$(eval excl+=$(if $(word $(a),$5),! -path $(word $(a),$5),)))
+$(eval all_headers := $(shell find $2 -name '*.hpp' $(excl)-o -name '*.cpp'))
```

Comment:
This changes supports up to 5 exclude options in the fifth argument. If the fifth argument is 'option1' 'option2' the find command is `shell find $2 -name '*.hpp' ! -path option1 ! -path option2 -o -name '*.cpp'`

Reply:
This has been resolved with a general foreach statement that now supports any number of exclude options as the fifth argument of the math-warnings function.